Python/Product/TestAdapter.Executor/Pytest/TestExecutorPytest.cs

```diff
@@ -132,22 +129,46 @@ IFrameworkHandle frameworkHandle
             IFrameworkHandle frameworkHandle
         ) {
             PythonProjectSettings settings = testGroup.Key;
-            if (settings.TestFramwork != TestFrameworkType.Pytest) {
+            if (settings == null || settings.TestFramwork != TestFrameworkType.Pytest) {
                 return;
             }
 
             using (var executor = new ExecutorService(settings, frameworkHandle, runContext)) {
+                Dictionary<string, TestResult> idToResultsMap = CreatePytestIdToVsTestResultsMap(testGroup);
                 var resultsXML = executor.Run(testGroup);
 
-                var testResults = TestResultParser.Parse(resultsXML, testGroup);
-                foreach (var result in testResults) {
+                //Read pytest results from xml
+                if (File.Exists(resultsXML)) {
+                    var xmlTestResultNodes = TestResultParser.Read(resultsXML).CreateNavigator().Select("/testsuite/testcase");
+                    foreach (XPathNavigator pytestResultNode in xmlTestResultNodes) {
+                        if (_cancelRequested.WaitOne(0)) {
+                            break;
+                        }
+                        try {
+                            string pytestId = TestResultParser.GetPytestId(pytestResultNode);
+                            if (pytestId != null && idToResultsMap.TryGetValue(pytestId, out TestResult vsTestResult)) {
+                                TestResultParser.UpdateVsTestResult(vsTestResult, pytestResultNode);
+                            } else {
+                                frameworkHandle.SendMessage(TestMessageLevel.Error, Strings.ErrorTestCaseNotFound.FormatInvariant(pytestResultNode.OuterXml));
```

Comment:
Should be FormatUI since that's for display. In this case, it's a string so it won't actually make a difference, but someone might take this as a coding example for a different data type where it makes a difference.

Reply:
I was thinking FormatInvariant would be better for when printing the raw xml.